test/e2e/validation.go

```diff
@@ -43,159 +46,112 @@ const (
+			By("VolumeSnapshotClass is ignored since VolumeSnapshotClasses are not supported in the given Cluster")
+			return nil
 		}
-		return err == nil
-	}, timeout, interval).Should(BeTrue())
-
-	debug("VolumeSnapshotClass found\n")
-	return nil
+		return err
+	}, timeout, interval).WithContext(ctx).Should(Succeed())
 }
 
 // function to validate CSI Driver.
-func validateCSIDriver(ctx context.Context) error {
-	cd := storagev1.CSIDriver{}
-
-	Eventually(func() bool {
-		err := crClient.Get(ctx, types.NamespacedName{Name: csiDriverName, Namespace: installNamespace}, &cd)
-		if err != nil {
-			debug("Error getting CSIDriver %s: %s\n", csiDriverName, err.Error())
-		}
-		return err == nil
-	}, timeout, interval).Should(BeTrue())
-
-	debug("CSIDriver found\n")
-	return nil
+func validateCSIDriver(ctx context.Context) bool {
+	By("validating the CSIDriver")
+	return Eventually(func(ctx context.Context) error {
+		return crClient.Get(ctx, types.NamespacedName{Name: csiDriverName, Namespace: installNamespace}, &storagev1.CSIDriver{})
+	}, timeout, interval).WithContext(ctx).Should(Succeed())
 }
 
 // function to validate TopoLVM node.
-func validateTopolvmNode(ctx context.Context) error {
-	ds := appsv1.DaemonSet{}
-	Eventually(func() bool {
-		err := crClient.Get(ctx, types.NamespacedName{Name: topolvmNodeDaemonSetName, Namespace: installNamespace}, &ds)
-		if err != nil {
-			debug("Error getting TopoLVM node daemonset %s: %s\n", topolvmNodeDaemonSetName, err.Error())
-		}
-		return err == nil
-	}, timeout, interval).Should(BeTrue())
-	debug("TopoLVM node daemonset found\n")
-
-	// checking for the ready status
-	Eventually(func() bool {
-		err := crClient.Get(ctx, types.NamespacedName{Name: topolvmNodeDaemonSetName, Namespace: installNamespace}, &ds)
-		if err != nil {
-			debug("Error getting TopoLVM node daemonset %s: %s\n", topolvmNodeDaemonSetName, err.Error())
-		}
-		return ds.Status.DesiredNumberScheduled == ds.Status.NumberReady
-	}, timeout, interval).Should(BeTrue())
-	debug("TopoLVM node pods are ready\n")
-
-	return nil
+func validateTopolvmNode(ctx context.Context) bool {
+	By("validating the TopoLVM Node DaemonSet")
+	return validateDaemonSet(ctx, types.NamespacedName{Name: topolvmNodeDaemonSetName, Namespace: installNamespace})
 }
 
 // function to validate vg manager resource.
-func validateVGManager(ctx context.Context) error {
-	ds := appsv1.DaemonSet{}
-
-	Eventually(func() bool {
-		err := crClient.Get(ctx, types.NamespacedName{Name: vgManagerDaemonsetName, Namespace: installNamespace}, &ds)
-		if err != nil {
-			debug("Error getting vg manager daemonset %s: %s\n", vgManagerDaemonsetName, err.Error())
-		}
-		return err == nil
-	}, timeout, interval).Should(BeTrue())
-	debug("VG manager daemonset found\n")
-
-	return nil
+func validateVGManager(ctx context.Context) bool {
+	By("validating the vg-manager DaemonSet")
+	return validateDaemonSet(ctx, types.NamespacedName{Name: vgManagerDaemonsetName, Namespace: installNamespace})
```

Comment:
Same comment about breaking it out

Reply:
Code duplication is never good, and both topoLVM and VG Manager share the same validation code for the daemon set readiness.